x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsPendingDeletionsIntegTests.java

```diff
@@ -0,0 +1,575 @@
+                        snapshots.add(response.getSnapshotInfo());
+                        snapshotLatch.countDown();
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        final AssertionError error = new AssertionError("error during snapshot", e);
+                        logger.error("test failed", error);
+                        snapshotLatch.countDown();
+                        throw error;
+                    }
+                });
+        }
+        snapshotLatch.await();
+
+        final int nbMounts = randomIntBetween(1, 10);
+        final CountDownLatch mountLatch = new CountDownLatch(nbMounts);
+        final Map<String, SnapshotId> mounts = ConcurrentCollections.newConcurrentMap();
+
+        for (int i = 0; i < nbMounts; i++) {
+            final String mount = "mount-" + i;
+
+            final SnapshotInfo snapshotInfo = randomFrom(snapshots);
+            client().execute(
+                MountSearchableSnapshotAction.INSTANCE,
+                new MountSearchableSnapshotRequest(
+                    mount,
+                    repository,
+                    snapshotInfo.snapshot().getSnapshotId().getName(),
+                    randomFrom(snapshotInfo.indices()),
+                    Settings.builder().put(SEARCHABLE_SNAPSHOTS_DELETE_SNAPSHOT_ON_INDEX_DELETION, true).build(),
+                    Strings.EMPTY_ARRAY,
+                    true,
+                    randomFrom(MountSearchableSnapshotRequest.Storage.values())
+                ),
+                new ActionListener<>() {
+                    @Override
+                    public void onResponse(RestoreSnapshotResponse response) {
+                        assertThat(response.getRestoreInfo().successfulShards(), greaterThan(0));
+                        assertThat(response.getRestoreInfo().failedShards(), equalTo(0));
+                        mounts.put(mount, snapshotInfo.snapshotId());
+                        mountLatch.countDown();
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        final AssertionError error = new AssertionError("error during mount", e);
+                        logger.error("test failed", error);
+                        mountLatch.countDown();
+                        throw error;
+                    }
+                }
+            );
+        }
+        mountLatch.await();
+
+        if (randomBoolean()) {
+            // we can't delete a repository that is used by searchable snapshot indices but we can update it
+            assertAcked(
+                clusterAdmin().preparePutRepository(repository)
+                    .setType(FsRepository.TYPE)
+                    .setVerify(randomBoolean())
+                    .setSettings(randomBoolean() ? repositorySettings : repositorySettings.put("dummy", randomInt()))
+            );
+        }
```

Comment:
I am not exactly sure what this bit validates? Perhaps add a comment?

Reply:
I added a comment in 2d8b32a649e777a273253716bbaccc8303f462fe:
>            // Force the re-registration of the repository with randomized value for the "verify" flag;
>             // it helps to test the case where the repository UUID is unknown at the time the snapshot
>             // is marked to be deleted.